.github/workflows/deploy.yml

```diff
@@ -0,0 +1,102 @@
+name: Deploy
+
+on:
+  push:
+    branches: [ 'develop', 'master' ]
+  pull_request:
+    branches: [ 'develop', 'master' ]
+
+jobs:
+  cancel-previous:
+    name: 'Cancel Previous Runs'
+    runs-on: ubuntu-latest
+    timeout-minutes: 3
+    steps:
+      - uses: styfle/cancel-workflow-action@0.8.0
+        with:
+          access_token: ${{ github.token }}
+
+  deploy:
+    if: "!contains(github.event.head_commit.author.name, 'GITHUB_ACTION')"
+    runs-on: ubuntu-latest
+    needs: cancel-previous
+    permissions:
+      contents: write
+      deployments: write
+      packages: write
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+          token: ${{ secrets.GH_TOKEN }}
+      - uses: actions/setup-node@v2-beta
+        with:
+          node-version: '16.10.0'
+          registry-url: https://registry.npmjs.org/
+
+      - name: Configure Git
+        run: |
+          git config --global user.email ${{ secrets.GH_EMAIL }}
+          git config --global user.name ${{ secrets.GH_USER }}
+          git remote set-url origin "https://${{ secrets.GH_TOKEN }}@github.com/energywebfoundation/iam-roles.git" > /dev/null 2>&1
+          git reset --hard
+          git fetch -u origin master:master
+          git fetch -u origin develop:develop
+
+      - name: Set npm credentials
+        run: npm config set //registry.npmjs.org/:_authToken ${{ secrets.NPM_TOKEN }}
+
+      - name: Install package
+        run: |
+          npm ci
+          npm run setup
+
+      - name: Build package
+        run: npm run build
+
+      - name: Npm run test
+        run: npm run test
+
+      - name: Output Run number
+        run: echo ${{ github.run_number }}
+
+      - name: Deploy on develop branch
+        if: github.ref == 'refs/heads/develop'
+        env:
+          GITHUB_TOKEN: ${{ secrets.GH_TOKEN }}
+          NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
+          RUN_NUMBER: ${{ github.run_number }}
+        run: |
+          npm run publish:canary
+
+      - name: Deploy on master branch
+        if: github.ref == 'refs/heads/master'
+        env:
+          GITHUB_TOKEN: ${{ secrets.GH_TOKEN }}
+          NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
+        run: |
+          git config user.name "GITHUB_ACTION"
+          npm run publish:release
+        
+  sync_master_to_develop_branch :
+    if: github.ref == 'refs/heads/master'
+    runs-on: ubuntu-latest
+    needs: deploy
+    permissions:
+      contents: write
+      deployments: write
+      packages: write
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+          ref: develop
+          token: ${{ secrets.GH_TOKEN }}
+
+      - name: Commit to develop branch and merge to master branch
+        run: |
+          git config --global user.email ${{ secrets.GH_EMAIL }}
```

Comment:
I am not sure if we need to set email and name since `develop` has already been checkout with `GH_TOKEN`

Reply:
A token is associated with a user and they might have different emails / orgs, so this is needed IMO.